spec/spec_helper.rb

```diff
@@ -14,7 +15,31 @@
   c.module_path = File.join(fixture_path, 'modules')
   c.manifest_dir = File.join(fixture_path, 'manifests')
   c.pattern = FileList[c.pattern].exclude(/^spec\/fixtures/)
+
+  # ensure specs are cleaned up
+  tmpdir = Dir.mktmpdir("rspecrun_dsc")
+  oldtmpdir = Dir.tmpdir()
+  ENV['TMPDIR'] = tmpdir
+
+  c.after :suite do
+    # return to original tmpdir
+    ENV['TMPDIR'] = oldtmpdir
+    if Puppet::Util::Platform.windows?
+      #FileUtils.rm_rf(tmpdir) - follows symlinks on Windows
+      # and deletes there as well :/
+      system("rmdir /s /q \"#{tmpdir}\"")
```

Comment:
Either way, this may need a pure verification. The symlinks are created calling `Puppet::FileSystem.symlink`, which calls `CreateSymbolicLinkW` - https://github.com/puppetlabs/puppet/blob/b3d9c03004bcb09d2539d48db666a2921f9bc481/lib/puppet/util/windows/file.rb#L82-L90